ferry.go

```diff
@@ -646,7 +653,7 @@ func (f *Ferry) Run() {
 	binlogWg.Wait()
 
 	f.logger.Info("ghostferry run is complete, shutting down auxiliary services")
-	f.OverallState = StateDone
+	f.OverallState.Store(StateDone)
```

Comment:
Not particularly related to your PR, but I noticed we are setting `StateDone` here, but it isn't actually until well after that we actually post the cutover unlock (in `sharding`, not in `copydb`).

Reply:
`StateDone` signals that ghostferry is done, and shop mover cutover unlock signals the shop move is done?